### PR TITLE
Preparing for containerization of Syncron

### PR DESCRIPTION
**Pull request by IgnacioPro**
This PR introduces a Dockerfile that will build the application when used in a container inside of Openshift.

### CODE REVIEW

**Comment by jsanemet** [comment]
You will have to update the module's name on 'go.mod' in order for things to work: https://github.com/RedHatCRE/syncron/blob/d17d441043fdc1ebc7fadf84209a0a56cf492d20/go.mod#L1

**Comment by jsanemet** [comment]
Out of curiosity, why is the previous name invalid?

**Comment by jsanemet** [comment]
Additionally, I gather that you would be more interested on the server mode of syncron for the docker image. Although that can be changed later on.

**Comment by IgnacioPro** [comment]
> Out of curiosity, why is the previous name invalid?

When performing `go install ` command the modules are not consistent.

`module declares its path as: github.com/rhcre/syncron
	    but was required as: github.com/redhatcre/syncron`

**Comment by jsanemet** [comment]
I think I see what is going on. Through that 'go install' command you are trying to download from the remote storage, like in PiPy. Syncron is not there yet, so if you want to install it you will need to clone the repository and do something like: `go install ./syncron/cmd/adhoc`

File: build/package/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM golang:1.18
+
+RUN git -c http.sslVerify=false clone https://github.com/RedHatCRE/syncron.git
+WORKDIR /go/syncron/
+RUN touch syncron.yaml
+RUN go mod tidy
+RUN go install cmd/adhoc/main.go
```
Comment (jsanemet) [inline build/package/Dockerfile:7]: FYI This command will install the binary under: '~/go/bin' as 'adhoc' without executable rights. You will probably need to both rename it and apply some chmod to it. That is at least what I had to do on my machine, Docker may be different.

**Review by jsanemet — APPROVED** [review]
Looks good to go. Although, are you not missing a 'CMD' step on the Dockerfile?. It does nothing right now.

**Comment by IgnacioPro** [comment]
> Looks good to go. Although, are you not missing a 'CMD' step on the Dockerfile?. It does nothing right now.

True, this was to get it installed with dependencies first.

**Review by adrianfusco — APPROVED** [review]
+1

**Pull request merged**